File: Phase1_High_Performance_Web_Service_for_Data_Retrieval/temp/saveFiles.py
#!/usr/bin/env python

import sys
from sys import argv
import os
import string
import gc

reload(sys)
sys.setdefaultencoding('utf-8')

def main():
	filename = "tweet-sort"
	f = open(filename, 'r')

	colon = ":"
	enter = "\n"
	space = "_"

	i = 0

	while 1:
		line = str(f.readline())
		if not line:
			break

		array = line.split("\t")

		if len(array) < 6:
			continue

		for j in range(5, len(array)):
			
			name = array[3].strip() + space + array[j].strip()
			fw = open(name, 'a')

			text = removeSimbols(array[4].strip())

			fw.write(array[0].strip() + colon + array[1].strip() + colon + array[2].strip() + colon + text + enter)

			fw.close()

		if i % 10000 == 0:
			gc.collect()

		i += 1


def removeSimbols(text):
	return text.\
			replace('\\"', '\"').\
            replace('\\v', '\v').\
            replace('\\t', '\t').\
            replace('\\r', '\r').\
            replace('\\n', '\n').\
            replace('\\f', '\f').\
            replace('\\b', '\b').\
            replace('\\a', '\a')

if __name__ == "__main__":
	main()

File: Phase1_High_Performance_Web_Service_for_Data_Retrieval/temp/saveFiles.java
import java.io.*;
import java.util.*;

public class saveFiles {

    public static void main(String args[]) throws IOException {

        // Read data
        File inFile = new File("tweet-sort");
        
        // If file doesnt exists, then create it
        if (!inFile.exists()) {
            System.err.println("No file called: " + "tweet");
            System.exit(-1);
        }

        BufferedReader br = null;

        // Read string from the input file
        String currentLine;
        String space = "_";
        String encode = "UTF8";
        String colon = ":";
        String enter = "\n";
        
        br = new BufferedReader(new FileReader(inFile));
        int i = 0;
        while ((currentLine = br.readLine()) != null) {
            
            String[] array = currentLine.split("\t");
            
            // Ignore data without hashtags
            if (array.length <= 5) {
                continue;
            }

            // String value = array[0];
            for (int j = 5; j < array.length; j++) {
                
                String name = array[3] + space + array[j];
                
                File fout = new File(name);

                if (!fout.exists()) {
                    fout.createNewFile();
                }

                BufferedWriter out = new BufferedWriter(new OutputStreamWriter(new FileOutputStream(fout, true), encode));

                out.write(array[0] + colon + array[1] + colon + array[2] + colon + array[4] + enter);
                out.flush();
                out.close();
            }

            if (i % 10000 == 0) {
                System.gc();
                System.out.println(i);
            } 
            i++;
        }
    }
}